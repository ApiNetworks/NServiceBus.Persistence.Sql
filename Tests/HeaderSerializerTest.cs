using System.Collections.Generic;
using ApprovalTests;
using NUnit.Framework;
using ObjectApproval;

public class HeaderSerializerTest
{
    [Test]
    public void ToXml()
    {
        var dictionary = new Dictionary<string, string> {{"TheKey", "TheValue"}};
        var xml = HeaderSerializer.ToXml(dictionary);
        Approvals.VerifyXml(xml);
 
    }

    [Test]
    public void FromXml()
    {
        var xml = @"<Headers>
  <Header>
    <Key>TheKey</Key>
    <Value>TheValue</Value>
  </Header>
</Headers>";
        ObjectApprover.VerifyWithJson(HeaderSerializer.FromString(xml));
    }
}