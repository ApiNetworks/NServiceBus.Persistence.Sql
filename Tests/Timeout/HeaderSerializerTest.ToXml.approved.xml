<Headers>
  <Header>
    <Key>TheKey</Key>
    <Value>TheValue</Value>
  </Header>
</Headers>